src/test/rpc/Handler_test.cpp

```diff
@@ -0,0 +1,133 @@
+{
+    return (os << ns.count() << "ns");
+}
+
+// NOTE This is a rather naiive effort at a microbenchmark. Ideally we want
+// Google Benchmark, or something similar. Also, this actually does not belong
+// to unit tests, as it makes little sense to run it in conditions very
+// dissimilar to how rippled will normally work.
+// TODO as https://github.com/XRPLF/rippled/issues/4765
+
+class Handler_test : public beast::unit_test::suite
+{
+    auto
+    time(std::size_t n, auto f, auto prng) -> auto
+    {
+        using clock = std::chrono::steady_clock;
+        assert(n > 0);
+        double sum = 0;
+        double sum_squared = 0;
+        std::size_t j = 0;
+        while (j < n)
+        {
+            // Generate 100 inputs upfront, separated from the inner loop
+            std::array<decltype(prng()), 100> inputs = {};
+            for (auto& i : inputs)
+            {
+                i = prng();
+            }
+
+            // Take 100 samples, then sort and throw away 35 from each end,
+            // using only middle 30. This helps to reduce measurement noise.
+            std::array<long, 100> samples = {};
+            for (std::size_t k = 0; k < 100; ++k)
+            {
+                auto start = std::chrono::steady_clock::now();
+                f(inputs[k]);
+                samples[k] = (std::chrono::steady_clock::now() - start).count();
+            }
+
+            std::sort(samples.begin(), samples.end());
+            for (std::size_t k = 35; k < 65; ++k)
+            {
+                j += 1;
+                sum += samples[k];
+                sum_squared += (samples[k] * samples[k]);
+            }
+        }
+
+        const double mean_squared = (sum * sum) / (j * j);
+        return std::make_tuple(
+            clock::duration{static_cast<long>(sum / j)},
+            clock::duration{
+                static_cast<long>(std::sqrt((sum_squared / j) - mean_squared))},
+            j);
+    }
+
+    void
+    reportLookupPerformance()
+    {
+        testcase("Handler lookup performance");
+
+        std::random_device dev;
+        std::ranlux48 prng(dev());
+
+        std::set<const char*> handlerNames = getHandlerNames();
+        std::vector<const char*> names(
+            handlerNames.begin(), handlerNames.end());
+
+        std::uniform_int_distribution<std::size_t> distr{0, names.size() - 1};
+
+        std::size_t dummy = 0;
+        auto const [mean, stdev, n] = time(
+            1'000'000,
+            [&](std::size_t i) {
+                auto const d = getHandler(1, false, names[i]);
+                dummy = dummy + i + (int)d->role_;
+            },
+            [&]() -> std::size_t { return distr(prng); });
+
+        std::cout << "mean=" << mean << " stdev=" << stdev << " N=" << n
+                  << '\n';
+
+        BEAST_EXPECT(dummy != 0);
+    }
+
+public:
+    void
+    run() override
+    {
+        reportLookupPerformance();
+    }
+};
+
+BEAST_DEFINE_TESTSUITE(Handler, rpc, ripple);
```

Comment:
Since this is testing performance timing and not really correctness, I think it could be changed to a manual test.
```
BEAST_DEFINE_TESTSUITE_MANUAL(Handler, rpc, ripple);
```

Reply:
Oh great, wan't aware of that macro